### ISSUE TITLE
DataGrid show vertical/horizontal scrollbars when "no result found" and while loading. 

### ISSUE THREAD

**Issue by boldurean** [mui/mui-x#8689]
### Duplicates

- [X] I have searched the existing issues

### Latest version

- [X] I have tested the latest version

### Steps to reproduce 🕹

Link to live example: https://react-ts-plvro3.stackblitz.io
Link to App: https://stackblitz.com/edit/react-ts-plvro3?file=App.tsx

Steps:
1. Make sure you click "Open in new tab" in order to reproduce it correctly
2. Fill the quick filter field in the toolbar with "gggg" _(value which doesn't exist in the rows)_  in order to get "no results found" message
3. see both scrollbars appear: 

<img width="2279" alt="Screenshot 2023-04-20 at 12 06 56" src="https://user-images.githubusercontent.com/64642760/233348330-d90a8140-7c07-4f44-bf65-59e701a2c398.png">

### Current behavior 😯

This scrollbars appear not only when no results found, they're also seen when DataGrid has "loading" state on. 
<img width="2269" alt="Screenshot 2023-04-20 at 12 09 56" src="https://user-images.githubusercontent.com/64642760/233348911-9e3a9898-7909-42c9-b8bc-621c6cfe5a8d.png">

It doesn't happen on the latest Mui-x/5+ versions. 

### Expected behavior 🤔

Scrollbars shouldn't be displayed for no reason. 

### Context 🔦

Most of the DataGrids in the project are used in the container which takes 100% width, and dynamic % of height depending of the screen. But looks like something wrong with inner Grid w/h calculation. 

### Your environment 🌎

<details>
  <summary><code>npx @mui/envinfo</code></summary>

```
  System:
    OS: macOS 13.3.1
  Binaries:
    Node: 19.9.0 - ~/.nvm/versions/node/v19.9.0/bin/node
    Yarn: 1.22.19 - ~/Library/pnpm/yarn
    npm: 9.6.3 - ~/.nvm/versions/node/v19.9.0/bin/npm
  Browsers:
    Chrome: 112.0.5615.137
    Edge: 112.0.1722.54
    Firefox: Not Found
    Safari: 16.4
  npmPackages:
    @emotion/react: ^11.10.6 => 11.10.6
    @emotion/styled: ^11.10.6 => 11.10.6
    @mui/icons-material: ^5.11.16 => 5.11.16
    @mui/lab: 5.0.0-alpha.114 => 5.0.0-alpha.114
    @mui/material: ^5.12.1 => 5.12.1
    @mui/x-data-grid: ^6.2.0 => 6.2.0
    @mui/x-date-pickers: ^6.2.0 => 6.2.0
    react: ^18.2.0 => 18.2.0
    react-dom: ^18.2.0 => 18.2.0
```
</details>

**Comment by boldurean** [comment]
Checked again [v5.17.24](https://github.com/mui/mui-x/releases/tag/v5.17.24) behaves normally.

**Comment by boldurean** [comment]
> Related to https://github.com/mui/mui-x/issues/6970
> 
> 
> 
> The horizontal scrollbar makes sense to be visible when the resolution is smaller than the sum of all column widths. The problem is that the scrollbar reduces the area available for the overlay, which has size equals to the main content size, causing vertical scrollbars too.

Yes i'm not talking about situations when container width is smaller that min-width settled for  columns. Then it make sense as well as in case of having 100 rows, it makes sense to see the vertical scrollbar. But not in the case described above.

**Comment by boldurean** [comment]
This issue isn't resolved btw in version 6.5.0

**Comment by boldurean** [comment]
@cherniavskii is there any chance to tag this as a bug?

**Comment by cherniavskii** [comment]
Hey @boldurean 
Yeah, I think the vertical scrollbar is not expected in this case

**Comment by boldurean** [comment]
> 

There is a issue with the overlay I guess it's size doesn't count the agent's scrollbar's which causing both scrollbars to appear with no reason :|

**Comment by cherniavskii** [comment]
I managed to create this minimal reproduction example: https://codesandbox.io/s/falling-water-mk01wv?file=/demo.tsx
The issue is only reproducible when the grid height has a subpixel value (like `500.5px`), which could often happen when using % or vh.

**Comment by romgrk** [comment]
Isn't this related to #9236? Might be fixed by that one.

**Comment by cherniavskii** [comment]
@romgrk Yes, thanks for linking it!

**Comment by boldurean** [comment]
Thanks folks for your attention to this. We cannot upgrade to further versions only because of this bug <3

**Comment by cherniavskii** [comment]
Hey @boldurean,
The pull request with a fix was merged shortly after the release of 6.8.0, therefore it wasn't included in it.
The fix will be included in this week's release (it usually happens on Thursday).
Sorry for the confusion!

**Comment by boldurean** [comment]
> Hey @boldurean,
> The pull request with a fix was merged shortly after the release of 6.8.0, therefore it wasn't included in it.
> The fix will be included in this week's release (it usually happens on Thursday).
> Sorry for the confusion!

Yay! Thanks for the answer! Great job <3
